diff --git a/src/Bedrock_DeFi_exp/Vault.sol b/src/Bedrock_DeFi_exp/Vault_patch.sol
index becc2c0..a673d7f 100644
--- a/src/Bedrock_DeFi_exp/Vault.sol
+++ b/src/Bedrock_DeFi_exp/Vault_patch.sol
@@ -2365,7 +2365,7 @@ abstract contract PausableUpgradeable is Initializable, ContextUpgradeable {
 
 // File: contracts/contracts/Vault.sol
 
-contract Vault is Initializable, AccessControlUpgradeable, PausableUpgradeable, ReentrancyGuardUpgradeable {
+contract Vault_patch is Initializable, AccessControlUpgradeable, PausableUpgradeable, ReentrancyGuardUpgradeable {
     bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
     bytes32 public constant OPERATOR_ROLE = keccak256("OPERATOR_ROLE");
 
@@ -2399,6 +2399,8 @@ contract Vault is Initializable, AccessControlUpgradeable, PausableUpgradeable,
      */
     function mint() external payable {
         require(!paused[NATIVE_BTC], "SYS002");
+        (, uint256 uniBTCAmount) = _amounts(msg.value);
+        require(uniBTCAmount * 1e10 < msg.value, "SYS008: Exchange rate unsafe - prevents 1:1 conversion");
         _mint(msg.sender, msg.value);
     }
 
