diff --git a/src/Bedrock_DeFi_exp/Vault.sol b/src/Bedrock_DeFi_exp/Vault_patch.sol
index becc2c0..aeddea9 100644
--- a/src/Bedrock_DeFi_exp/Vault.sol
+++ b/src/Bedrock_DeFi_exp/Vault_patch.sol
@@ -2364,8 +2364,15 @@ abstract contract PausableUpgradeable is Initializable, ContextUpgradeable {
 }
 
 // File: contracts/contracts/Vault.sol
-
-contract Vault is Initializable, AccessControlUpgradeable, PausableUpgradeable, ReentrancyGuardUpgradeable {
+/**
+ * Patch: 
+ * Added a require() in the mint function 
+ * (`require(msg.value < 1e15, "SYS007: ETH-to-uniBTC minting disabled due to unsafe exchange rate"`);
+ * to disable minting with native ETH when the sent amount exceeds a very small threshold (1e15 wei).
+ * This prevents abuse from unsafe exchange rates during ETH-to-uniBTC conversion, 
+ * mitigating the risk of price exploitation attacks.
+ */
+contract Vault_patch is Initializable, AccessControlUpgradeable, PausableUpgradeable, ReentrancyGuardUpgradeable {
     bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
     bytes32 public constant OPERATOR_ROLE = keccak256("OPERATOR_ROLE");
 
@@ -2399,6 +2406,7 @@ contract Vault is Initializable, AccessControlUpgradeable, PausableUpgradeable,
      */
     function mint() external payable {
         require(!paused[NATIVE_BTC], "SYS002");
+        require(msg.value < 1e15, "SYS007: ETH-to-uniBTC minting disabled due to unsafe exchange rate");
         _mint(msg.sender, msg.value);
     }
 
